Services/CustomsService.cs

```diff
@@ -0,0 +1,265 @@
+using CustomsController.Model;
+using Microsoft.EntityFrameworkCore;
+
+namespace CustomsController.Services;
+
+/// <inheritdoc/>
+public class CustomsService : ICustomsService
+{
+    private readonly CustomsContext _customContext;
+
+    /// <summary>
+    /// Constructor
+    /// </summary>
+    /// <param name="customsContext">for logic</param>
+    public CustomsService(CustomsContext customsContext)
+    {
+        _customContext = customsContext;
+    }
+
+    /// <inheritdoc/>
+    public Country AddNewCountry(string A2Code, bool isEUCU)
+    {
+        var country = new Country(A2Code, isEUCU);
+        var code = _customContext.Countries.FirstOrDefault(c => c.A2Code == A2Code);
+        if (code != default)
+        {
+            return default;
+        }
+        _customContext.Add(country);
+        _customContext.SaveChanges();
+        return country;
+    }
+
+    /// <inheritdoc/>
+    public Country RemoveCountry(string A2Code)
+    {
+        var country = _customContext.Countries.FirstOrDefault(c => c.A2Code == A2Code);
+        if (country == default)
+        {
+            return null;
+        }
+        _customContext.Remove(country);
+        _customContext.SaveChanges();
+        return country;
+    }
+
+    /// <inheritdoc/>
+    public string ChangeEUCU(string A2Code, bool val)
+    {
+        var country = _customContext.Countries.FirstOrDefault(c => c.A2Code == A2Code);
+        if (country != default)
+        {
+            country.IsEUCU = val;
+            _customContext.SaveChanges();
+            return $"{val}";
+        }
+        else
+        {
+            return "country is not in the database";
+        }
+    }
+
+    /// <inheritdoc/>
+    public string GetCountryEUCU(string A2Code)
+    {
+        var country = _customContext.Countries.FirstOrDefault(c => c.A2Code == A2Code);
+        if (country != default)
+        {
+            return $"{country.IsEUCU}";
+        }
+        else
+        {
+            return "country is not in the database";
+        }
+
+    }
+
+    /// <inheritdoc/>
+    public List<Country> GetAllCountries()
+    {
+        return _customContext.Countries.ToList();
+    }
+
+    /// <inheritdoc/>
+    public CustomsResponse GetCustoms(string country1code, string country2code)
+    {
+        var country1_isEUCU = _customContext.Countries.FirstOrDefault(c => c.A2Code == country1code).IsEUCU;
+        var country2_isEUCU = _customContext.Countries.FirstOrDefault(c => c.A2Code == country2code).IsEUCU;
+
+        var doesCountryExist = DoesCountryExist(country1code, country2code);
+        if (doesCountryExist != null)
+            return doesCountryExist;
+
+        var result2 = new CustomsResponse()
+        {
+            IfCustomsInEUCU = !(country1_isEUCU && country2_isEUCU),
+            Success = true
+        };
+        return result2;
+    }
+
+    private CustomsResponse DoesCountryExist(string country1code, string country2code)
```

Comment:
If we phrase the function name like a y/n-question we should also return a bool and not an object. You can either rename the function or change the response to a bool and work with that

Reply:
ok